### DISCUSSION TITLE
PerlMagick Installation

### DISCUSSION THREAD

**Discussion by ghunt8510** [ImageMagick/ImageMagick#2607] [Help]
I have installed ImageMagick on my Windows 10 machine and it seems to work.  I can run command line scripts.  I need to use ImageMagick inside of my Perl script, so I need PerlMagick.  I selected the checkbox on the ImageMagick install to install PerlMagick.  It seems to not have worked.  I see in the installed directory(C:\Program Files\ImageMagick-7.0.10-Q16-HDRI\PerlMagick) that there is a readme.txt that says to run the Image-Magick.ppd file if PerlMagick didn't seem to install.  When I run this file, it says "PPM functionality is being replaced by the State Tool, which does things differently.".  Can anyone help me figure out how to get PerlMagick installed on my Windows 10 machine, so I can use ImageMagick in my Perl scripts?  This shouldn't be this hard.....

C:\Program Files\ImageMagick-7.0.10-Q16-HDRI\PerlMagick\demo>perl demo.pl
Can't locate loadable object for module Image::Magick in @INC (@INC contains: C:/Perl64/site/lib C:/Perl64/lib) at demo.pl line 5.
Compilation failed in require at demo.pl line 5.
BEGIN failed--compilation aborted at demo.pl line 5.

**Comment by t246246** [comment]
From your error message, I guess you installed ActivePerl, in which PPM is no longer available.
https://www.activestate.com/blog/goodbye-ppm-hello-state-tool/
I don't know how to use it.
Instead you can use Strawberry perl, which still has ppm.
https://stackoverflow.com/questions/6643939/installing-modules-using-strawberry-perl

I tried ImageMagick-7.0.10-31-Q16-HDRI-x64-dll.exe and strawberry-perl-5.30.3.1-64bit.msi, and got success.

c:\Strawberry\perl\bin>perl -MImage::Magick -e "print $Image::Magick::VERSION"
6.89

However with latest strawberry-perl-5.32.0.1-64bit.msi, it failed.

C:\Strawberry\perl\bin>ppm
PPM interactive shell (11.11_04) - type 'help' for available commands.
PPM> install Image::Magick
Install package 'Image-Magick?' (y/N): y
Installing package 'Image-Magick'...
Error installing package 'Image-Magick': Read a PPD for 'Image-Magick', but it is not intended for this build of Perl (MSWin32-x64-multi-thread-5.32)
